### DISCUSSION TITLE
createPage() with ?__contentFilePath in component causes 'Multiple "root" queries found' in build when the component has a named query

### DISCUSSION THREAD

**Discussion by adam1658** [gatsbyjs/gatsby#36420] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I'm using `createPage` within `createPages` in `gatsby-node.js`.

Because these pages are for mdx with `gatsby-plugin-mdx`, I specify `component` as follows:
```js
createPage({
  path: mdxSlug,
  component: `${template}?__contentFilePath=`${mdxPath}`,
  context: {slug: mdxSlug},
})
```
(see https://github.com/gatsbyjs/gatsby/blob/dc283d7a10a0437275dabdbc2bcec60b4cff6104/packages/gatsby-plugin-mdx/README.md?plain=1#L243)

In addition to this, I name the query in the referenced component so that I can generate TS types (I set `graphqlTypegen: true,` in `gatsby-config.js`, and only named queries get types generated.

HOWEVER, now I'm getting the 'Multiple "root" queries found' error in the `gatsby develop` output, because that query now appears in more than one 'component' (probably because that one component is referenced multiple times with different names due to the dynamic `__contentFilePath=...`).

My repo below does not use gatsby-plugin-mdx or generate types, but does use `createPage` twice with the same component but different `__contentFilePath`s, and has a named query in the referenced component, to help isolate the issue. 

### Reproduction Link

https://github.com/adam1658/gatsby-repro-multiple-queries-querystring

### Steps to Reproduce

1. Clone the repo
2. `npm install`
3. `npm develop`
4. see errors

### Expected Result

The one component (DemoPage in `src/templates/Demo.js`)  has one dynamic query with one name (`DemoPage`), therefore there is only one query in the whole application 

### Actual Result

Gatsby complains that there are two queries in the whole application with the same name (`DemoPage`)

### Environment

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 17.8.0 - ~/.nodenv/versions/17.8.0/bin/node
    npm: 8.5.5 - ~/.nodenv/versions/17.8.0/bin/npm
  Languages:
    Python: 3.8.10 - /usr/bin/python
  Browsers:
    Chrome: 103.0.5060.114
  npmPackages:
    gatsby: ^4.21.0 => 4.21.0
```

**Comment by LekoArts** [comment, marked as answer]
Hi!

Please see [#25068 (comment)](https://github.com/gatsbyjs/gatsby/discussions/25068?sort=new#discussioncomment-3295280). This current limitation is noted in the README (https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#updating-page-templates) and after your comment here I also added it to the guide (https://www.gatsbyjs.com/docs/how-to/routing/mdx/#make-a-layout-template-for-your-posts).

So at the moment it's working as intended, but eventually we'll fix this limitation, I made a note of that. We'll announce that fix in the release notes (and remove the old notes then)

**Reply by adam1658** [reply]
Thanks @LekoArts, I appreciate your help
